src/libraries/System.Private.CoreLib/src/System/Diagnostics/CodeAnalysis/NullableAttributes.cs

```diff
@@ -125,4 +125,55 @@ sealed class DoesNotReturnIfAttribute : Attribute
         /// <summary>Gets the condition parameter value.</summary>
         public bool ParameterValue { get; }
     }
+
+    /// <summary>Specifies that the method or property will ensure that the listed field and property members have not-null values.</summary>
+    [AttributeUsage(AttributeTargets.Method | AttributeTargets.Property, Inherited = false)]
+#if INTERNAL_NULLABLE_ATTRIBUTES
+    internal
+#else
+    [CLSCompliant(false)]
+    public
+#endif
+        sealed class MemberNotNullAttribute : Attribute
+    {
+        /// <summary>Gets field or property member names.</summary>
+        public string[] Members { get; }
+
```

Comment:
Nit: please move the properties to below the ctor (here and in the other attribute as well)